pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -478,6 +486,15 @@ func (c *Cluster) provisionOSDContainer(devices []rookalpha.Device, selection ro
 	}
 }
 
+func (c *Cluster) skipVolumeForDirectory(path string) bool {
+	// If attempting to add both a directory and the dataDirHostPath of /var/lib/rook, we need to skip the volume and volume mount
+	// since it will have already been added for the dataDirHostPath
+	if path == c.dataDirHostPath && path == k8sutil.DataDir {
```

Comment:
not sure i understand the need for the second part of this conditional.  Why isn't `path = c.dataDirHostPath` sufficient?

also, do we have a general check somewhere else in the codebase to prevent overlapping directories being allowed as OSDs?

Reply:
i think i get it now.  so do we need the first part of the conditional at all then?  isn't it only a problem when the directory path == /var/lib/rook?

Reply:
nice catch, you're right we only need the check for /var/lib/rook. It's updated now.